drivers: clock_control: Move stm32f1 series to Cube LL API

erwango opened the pull request:
stm32f1 series migrated to Cube LL API.
This completes migration of stm32 clock control drivers to unique API and hence simplify generic
driver implementation in calls to clock control driver.
Former driver code cleaned up.

erwango committed:
drivers: clock_control: provide LL based driver to stm32f1xx series

Align stm32f1xx series clock driver to other parts of stm32 family.
Driver support both Connectivity and Density lines of stm32f1 series,
that are based on different Reset and Clock Control architectures.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: move stm32f1xx based boards to LL based Clock control driver

Following introduction of stm32cube LL based clock control driver
for stm32f1 series, update stm32f1xx based boards to support new
driver settings

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
stm32f1xx: remove references to native driver

Following introduction of stm32cube LL based clock control driver,
remove references to former native driver.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers: gpio stm32: set the ground for stm32f1x ll clock driver

Before introduction of LL based clock control for stm32f1xx series,
prepare gpio driver to provide useful definition for new API

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -222,11 +273,11 @@ config CLOCK_STM32_APB1_PRESCALER
 	  APB2 High speed clock (PCLK2) prescaler, allowed values:
 	  1, 2, 4, 8, 16
 
+endif #
```
end if #CLOCK_CONTROL_STM32_CUBE

erwango replied:
Done. Thanks.

galak reviewed (changes requested):
minor change to Kconfig comment

galak commented on arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.series:
```diff
@@ -12,6 +12,13 @@ source "arch/arm/soc/st_stm32/stm32f1/Kconfig.defconfig.stm32f1*"
 config SOC_SERIES
 	default stm32f1
 
+if CLOCK_CONTROL
```
Can we do:

select CLOCK_CONTROL_STM32_CUBE if CLOCK_CONTROL

in arch/arm/soc/st_stm32/Kconfig so we don't need to duplicate this everywhere

erwango replied:
ok, will do

galak reviewed (changes requested):
Some Kconfig possibly cleanup

galak reviewed (changes requested):
Should gpio before remove of the native ll change?

erwango committed:
stm32: clean up after completion of transition to ll Clock control

Following migration of stm32f1xx series clock control driver to
STM32Cube LL API, cleanup stm32 code base in order to take into
account that this is the only clock driver available for stm32
family.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
About GPIO:
It is needed to get gpio_stm32.h modif available when activating CONFIG_CLOCK_CONTROL_STM32_CUBE in in boards defconfig.

The gpio_stm32.c could be moved in clean up, but I decided to group .h and .c gpio stuff

galak commented:
For the gpio stuff, I just meant move the commit earlier in the series of commits before the board changes to enable the CUBE clock support

galak committed:
Merge branch 'arm' into stm32f1_ll_clock
